### DISCUSSION TITLE
option to change folder colors

### DISCUSSION THREAD

**Discussion by guillopacheco** [SylEleuth/gruvbox-plus-icon-pack#89] [Ideas]
Hello. I just installed these icon pack, its great thank you. I have an issue trying to change the folders, in gnome the default color for the folders are blue. How can I change the folders to the golden ones found inside the icon pack?

**Comment by guillopacheco** [comment]
i'm sorry, I kept looking and found this file browser extension, at least its not 1 folder at a time. https://github.com/costales/folder-color

**Comment by SylEleuth** [comment]
That set of scripts are only usable for specific file managers. There is a plan to make a way to change folders colors globally. Each file named `folder-*.svg` is actually a soft link to a specific file. Currently it links to `folder-blue-*.svg`, so it shouldn't be hard to just change that symlink path to a different folder with a different color. I can't make that kind of script right now but it should be in pack. Maybe someone else will find some time and do it.

**Comment by Clemovski** [comment]
Hello.

This is the quick and dirty script I have made to replace my folder colors. You just have to replace highland with the color of your choice, give it execution privileges, and launch it from ~/.icons/gruvbox-plus-icon-pack-5.5.0/places/scalable .
Hope it will help someone in the future.

Note that bookmark folders do not have color variants.

<details>
<summary>Click to expand</summary>
#!/usr/bin/env bash

rm certificate-server.svg
ln -s folder-highland-locked.svg certificate-server.svg
rm desktop.svg
ln -s user-highland-desktop.svg desktop.svg
rm document-open-recent.svg
ln -s folder-highland-recent.svg document-open-recent.svg
rm edittrash.svg
ln -s user-trash-full.svg edittrash.svg
rm emptytrash.svg
ln -s user-trash.svg emptytrash.svg
rm folder-activities.svg
ln -s folder-highland-activities.svg folder-activities.svg
rm folder-apple.svg
ln -s folder-highland-apple.svg folder-apple.svg
rm folder-apps.svg
ln -s folder-highland-apps.svg folder-apps.svg
rm folder-atom.svg
ln -s folder-highland-atom.svg folder-atom.svg
rm folder-camera.svg
ln -s folder-photo.svg folder-camera.svg
rm folder-cd.svg
ln -s folder-highland-cd.svg folder-cd.svg
rm folder-cinnamon.svg
ln -s folder-highland-cinnamon.svg folder-cinnamon.svg
rm folder-cloud.svg
ln -s folder-highland-mail-cloud.svg folder-cloud.svg
rm folder-decrypted.svg
ln -s folder-unlocked.svg folder-decrypted.svg
rm folder-desktop.svg
ln -s user-highland-desktop.svg folder-desktop.svg
rm folder-development.svg
ln -s folder-highland-development.svg folder-development.svg
rm folder-docker.svg
ln -s folder-highland-docker.svg folder-docker.svg
rm folder-documents-open.svg
ln -s folder-highland-documents-open.svg folder-documents-open.svg
rm folder-documents.svg
ln -s folder-highland-documents.svg folder-documents.svg
rm folder-download-open.svg
ln -s folder-highland-download-open.svg folder-download-open.svg
rm folder-downloads.svg
ln -s folder-highland-download.svg folder-downloads.svg
rm folder-download.svg
ln -s folder-highland-download.svg folder-download.svg
rm folder-drag-accept.svg
ln -s folder-highland-drag-accept.svg folder-drag-accept.svg
rm folder-dropbox.svg
ln -s folder-highland-dropbox.svg folder-dropbox.svg
rm folder-electron.svg
ln -s folder-highland-electron.svg folder-electron.svg
rm folder-encrypted.svg
ln -s folder-highland-locked.svg folder-encrypted.svg
rm folder-favorites.svg
ln -s folder-highland-favorites.svg folder-favorites.svg
rm folder-finder.svg
ln -s folder-highland-mac.svg folder-finder.svg
rm folder-fonts.svg
ln -s folder-highland-fonts.svg folder-fonts.svg
rm folder-games.svg
ln -s folder-highland-games.svg folder-games.svg
rm folder-gdrive.svg
ln -s folder-highland-google-drive.svg folder-gdrive.svg
rm folder-github.svg
ln -s folder-highland-github.svg folder-github.svg
rm folder-gitlab.svg
ln -s folder-highland-gitlab.svg folder-gitlab.svg
rm folder-git.svg
ln -s folder-highland-git.svg folder-git.svg
rm folder-gnome.svg
ln -s folder-highland-gnome.svg folder-gnome.svg
rm folder-google-drive.svg
ln -s folder-highland-google-drive.svg folder-google-drive.svg
rm folder-home.svg
ln -s user-home.svg folder-home.svg
rm folder_home.svg
ln -s user-home.svg folder_home.svg
rm folder-html.svg
ln -s folder-network.svg folder-html.svg
rm folder-icons.svg
ln -s folder-highland-icons.svg folder-icons.svg
rm folder-image-people.svg
ln -s folder-highland-image-people.svg folder-image-people.svg
rm folder-images.svg
ln -s folder-highland-pictures.svg folder-images.svg
rm folder-image.svg
ln -s folder-highland-pictures.svg folder-image.svg
rm folder-important.svg
ln -s folder-highland-important.svg folder-important.svg
rm folder-java.svg
ln -s folder-highland-java.svg folder-java.svg
rm folder-linux.svg
ln -s folder-highland-linux.svg folder-linux.svg
rm folder-locked.svg
ln -s folder-highland-locked.svg folder-locked.svg
rm folder-mac.svg
ln -s folder-highland-mac.svg folder-mac.svg
rm folder-mail-cloud.svg
ln -s folder-highland-mail-cloud.svg folder-mail-cloud.svg
rm folder-mail.svg
ln -s folder-highland-mail.svg folder-mail.svg
rm folder-mega.svg
ln -s folder-highland-mega.svg folder-mega.svg
rm folder-meocloud.svg
ln -s folder-highland-meocloud.svg folder-meocloud.svg
rm folder-mozilla.svg
ln -s folder-highland-mozilla.svg folder-mozilla.svg
rm folder-music-open.svg
ln -s folder-highland-music-open.svg folder-music-open.svg
rm folder-music.svg
ln -s folder-highland-music.svg folder-music.svg
rm folder-network.svg
ln -s folder-highland-network.svg folder-network.svg
rm folder-open-recent.svg
ln -s folder-recent.svg folder-open-recent.svg
rm folder-open.svg
ln -s folder-highland-open.svg folder-open.svg
rm folder_open.svg
ln -s folder-open.svg folder_open.svg
rm folder-owncloud.svg
ln -s folder-highland-owncloud.svg folder-owncloud.svg
rm folder-pcloud.svg
ln -s folder-highland-pcloud.svg folder-pcloud.svg
rm folder-photos.svg
ln -s folder-photo.svg folder-photos.svg
rm folder-photo.svg
ln -s folder-highland-photo.svg folder-photo.svg
rm folder-pictures-open.svg
ln -s folder-highland-pictures-open.svg folder-pictures-open.svg
rm folder-pictures.svg
ln -s folder-highland-pictures.svg folder-pictures.svg
rm folder-picture.svg
ln -s folder-highland-pictures.svg folder-picture.svg
rm folder-print.svg
ln -s folder-highland-print.svg folder-print.svg
rm folder-private.svg
ln -s folder-highland-private.svg folder-private.svg
rm folder-publicshare-open.svg
ln -s folder-highland-publicshare-open.svg folder-publicshare-open.svg
rm folder-publicshare.svg
ln -s folder-highland-image-people.svg folder-publicshare.svg
rm folder-public.svg
ln -s folder-publicshare.svg folder-public.svg
rm folder-recent.svg
ln -s folder-highland-recent.svg folder-recent.svg
rm folder-remote-ftp.svg
ln -s folder-remote.svg folder-remote-ftp.svg
rm folder-remote-nfs.svg
ln -s folder-remote.svg folder-remote-nfs.svg
rm folder-remote-open.svg
ln -s folder-highland-remote-open.svg folder-remote-open.svg
rm folder-remote-smb.svg
ln -s folder-remote.svg folder-remote-smb.svg
rm folder-remote-ssh.svg
ln -s folder-remote.svg folder-remote-ssh.svg
rm folder-remote.svg
ln -s folder-highland-remote.svg folder-remote.svg
rm folder-saved-search.svg
ln -s folder-highland-saved-search.svg folder-saved-search.svg
rm folder-script.svg
ln -s folder-highland-script.svg folder-script.svg
rm folder-sound.svg
ln -s folder-highland-music.svg folder-sound.svg
rm folder-steam.svg
ln -s folder-highland-steam.svg folder-steam.svg
rm folder.svg
ln -s folder-highland.svg folder.svg
rm folder-tar.svg
ln -s folder-highland-tar.svg folder-tar.svg
rm folder-templates-open.svg
ln -s folder-highland-templates-open.svg folder-templates-open.svg
rm folder-templates.svg
ln -s folder-highland-templates.svg folder-templates.svg
rm folder-temp.svg
ln -s folder-recent.svg folder-temp.svg
rm folder-text.svg
ln -s folder-highland-documents.svg folder-text.svg
rm folder-themes.svg
ln -s folder-highland-themes.svg folder-themes.svg
rm folder-torrent.svg
ln -s folder-highland-torrent.svg folder-torrent.svg
rm folder-txt.svg
ln -s folder-highland-documents.svg folder-txt.svg
rm folder-unlocked.svg
ln -s folder-highland-unlocked.svg folder-unlocked.svg
rm folder-vbox.svg
ln -s folder-highland-vbox.svg folder-vbox.svg
rm folder-videocamera.svg
ln -s folder-video.svg folder-videocamera.svg
rm folder-videos-open.svg
ln -s folder-highland-videos-open.svg folder-videos-open.svg
rm folder-videos.svg
ln -s folder-video.svg folder-videos.svg
rm folder-video.svg
ln -s folder-highland-video.svg folder-video.svg
rm folder-virtual.svg
ln -s folder-remote.svg folder-virtual.svg
rm folder-visiting.svg
ln -s folder-highland-visiting.svg folder-visiting.svg
rm folder-vscode.svg
ln -s folder-highland-vscode.svg folder-vscode.svg
rm folder-wifi.svg
ln -s folder-highland-wifi.svg folder-wifi.svg
rm folder-windows.svg
ln -s folder-highland-wine.svg folder-windows.svg
rm folder-wine.svg
ln -s folder-highland-wine.svg folder-wine.svg
rm folder-yandex-disk.svg
ln -s folder-highland-yandex-disk.svg folder-yandex-disk.svg
rm gnome-dev-trash-empty.svg
ln -s user-trash.svg gnome-dev-trash-empty.svg
rm gnome-dev-trash-full.svg
ln -s user-trash-full.svg gnome-dev-trash-full.svg
rm gnome-fs-directory-accept.svg
ln -s folder-open.svg gnome-fs-directory-accept.svg
rm gnome-fs-directory.svg
ln -s folder.svg gnome-fs-directory.svg
rm gnome-fs-home.svg
ln -s user-home.svg gnome-fs-home.svg
rm gnome-fs-trash-empty-accept.svg
ln -s user-trash-full.svg gnome-fs-trash-empty-accept.svg
rm gnome-fs-trash-empty.svg
ln -s user-trash.svg gnome-fs-trash-empty.svg
rm gnome-fs-trash-full.svg
ln -s user-trash-full.svg gnome-fs-trash-full.svg
rm gnome-home.svg
ln -s user-home.svg gnome-home.svg
rm gnome-stock-trash-empty.svg
ln -s user-trash.svg gnome-stock-trash-empty.svg
rm gnome-stock-trash-full.svg
ln -s user-trash-full.svg gnome-stock-trash-full.svg
rm gnome-stock-trash.svg
ln -s user-trash.svg gnome-stock-trash.svg
rm gtk-directory.svg
ln -s folder.svg gtk-directory.svg
rm inode-directory.svg
ln -s folder.svg inode-directory.svg
rm insync-folder.svg
ln -s folder-highland-google-drive.svg insync-folder.svg
rm knetattach.svg
ln -s folder-remote.svg knetattach.svg
rm library-music.svg
ln -s folder-highland-music.svg library-music.svg
rm network.svg
ln -s folder-network.svg network.svg
rm network-workgroup.svg
ln -s folder-highland-network.svg network-workgroup.svg
rm repository.svg
ln -s folder-network.svg repository.svg
rm stock_folder.svg
ln -s folder.svg stock_folder.svg
rm stock_open.svg
ln -s folder-open.svg stock_open.svg
rm stock_trash_empty.svg
ln -s user-trash.svg stock_trash_empty.svg
rm stock_trash_full.svg
ln -s user-trash-full.svg stock_trash_full.svg
rm trashcan_empty.svg
ln -s user-trash.svg trashcan_empty.svg
rm trashcan_full.svg
ln -s user-trash-full.svg trashcan_full.svg
rm user-desktop.svg
ln -s user-highland-desktop.svg user-desktop.svg
rm user-home-open.svg
ln -s user-highland-home-open.svg user-home-open.svg
rm user-home.svg
ln -s user-highland-home.svg user-home.svg
rm xfce-trash_empty.svg
ln -s user-trash.svg xfce-trash_empty.svg
rm xfce-trash_full.svg
ln -s user-trash-full.svg xfce-trash_full.svg
</details>

**Reply by acchs** [reply]
Thank you. The script is lovely.

**Comment by akarzim** [comment]
Hi,

Here is a more bash-ish script I've made to replace folders color.

<details><summary>Details</summary>
<p>

```sh
#!/bin/bash
set -euo pipefail
IFS=$'\n\t'

export FOLDER_COLOR="${FOLDER_COLOR:-"blue"}"

if [[ -f "folder-${FOLDER_COLOR}.svg" ]]; then
  for i in $(realpath "*-${FOLDER_COLOR}*.svg"); do
    filename="${i##*/}"
    ln -sfn "${filename}" "${filename/-${FOLDER_COLOR}/}"
  done
  echo "Done."
else
  echo "Invalid color: ${FOLDER_COLOR}"
  echo "Please peak one of:"
  find "${0%/*}" -type f -iname "folder-*-*.svg" | cut --delimiter "-" --fields 2 | sort | uniq | paste -sd " "
  exit 1
fi
```

</p>
</details> 

### Usage 

```sh
FOLDER_COLOR=violet ./folder-color-chooser.sh
```

If all goes well, it simply prints ‘Done’, if not, it prints a list of available colours and exits with an error code:
```sh
FOLDER_COLOR=coffee ./folder-color-chooser.sh
Invalid color: coffee
Please peak one of:
black blue citron firebrick gold green grey highland jade lavender lime olive orange pistachio pumpkin purple red rust sapphire tomato violet white yellow
```

Enjoy it!

**Reply by SylEleuth** [reply]
I haven't test it yet but for anyone using git version I would suggest to hold on until I finish uploading all folder colours (it shouldn't take longer than 2 weeks). There are some changes between different colours right now - some files are removed, some heavily changed and many added.

Few notes:
- default colour is `plasma` now
- I don't know if you took it under consideration but besides `folder-*colour*` files there are also `user-*colour*-desktop`, `user-*colour*-trash`, `user-*colour*-trash-full`, `bookmark-*colour*` and `bookmark-*colour*-missing`

**Reply by akarzim** [reply]
Thanks for the quick feedback !

> I don't know if you took it under consideration but besides folder-*colour* files there are also user-*colour*-desktop, user-*colour*-trash, user-*colour*-trash-full, bookmark-*colour* and bookmark-*colour*-missing

Yes, I do.

Here is an new version of my script with some enhancements : 
- defaults to `plasma` color
- rename `FOLDER_COLOR` to `PLACES_COLOR`
- adding a `--help` option to print a usage message
- adding a `--dry-run` option to simulate changes
- adding a `--verbose` option to know which file is touched
- adding a `--list` option to list (computed) available colors
- adding a `--color` option for those who prefer options to environment variables ;)

<details><summary>Code of places-color-chooser.sh</summary>
<p>

```sh
#!/bin/bash
set -euo pipefail
IFS=$'\n\t'

export PLACES_COLOR="${PLACES_COLOR:-"plasma"}"

colors() {
  find "${0%/*}" -type f -iname "folder-*-*.svg" | cut --delimiter "-" --fields 2 | sort | uniq | paste -sd " "
}

help="Places color chooser

Usage: ${0##*/} [-c | --color] PLACES_COLOR [-h | --help] [-l | --list] [-n | --dry-run] [-v | --verbose]

Environment:
  PLACES_COLOR  color to change to (default: plasma)

Options:
  -c, --color=PLACES_COLOR    set the new places color (default: plasma)
  -h, --help                  show this help
  -l, --list                  list available colors
  -n, --dry-run               perform a trial run with no changes made
  -v, --verbose               increase verbosity"

# options
while [[ "$#" -gt 0 && "$1" =~ ^- && ! "$1" == "--" ]]; do case "$1" in
  -c | --color )
    shift; PLACES_COLOR="$1"
    ;;
  -h | --help )
    echo -e "${help}"
    exit
    ;;
  -l | --list )
    echo "Available colors are:"
    echo "$(colors)"
    exit
    ;;
  -n | --dry-run )
    DRYRUN=1
    ;;
  -v | --verbose )
    VERBOSE=1
    ;;
esac; shift; done
if [[ "$#" -gt 0 && "$1" == '--' ]]; then shift; fi

DRYRUN="${DRYRUN:-0}"
VERBOSE="${VERBOSE:-0}"

if [[ -f "folder-${PLACES_COLOR}.svg" ]]; then
  for i in $(realpath "*-${PLACES_COLOR}*.svg"); do
    filename="${i##*/}"

    if [[ ${DRYRUN} -eq 1 || ${VERBOSE} -eq 1 ]]; then
      case "${filename}" in
        "bookmarks-${PLACES_COLOR}.svg")
          echo "${filename} → folder-bookmark.svg"
          ;;

        *)
          echo "${filename} → ${filename/-${PLACES_COLOR}/}"
          ;;
      esac
    fi

    if [[ ${DRYRUN} -eq 0 ]]; then
      case "${filename}" in
        "bookmarks-${PLACES_COLOR}.svg")
          ln -sfn "${filename}" "folder-bookmark.svg"
          ;;

        *)
          ln -sfn "${filename}" "${filename/-${PLACES_COLOR}/}"
          ;;
      esac
    fi
  done

  if [[ ${DRYRUN} -eq 1 ]]; then
    echo "Nothing done."
  else
    echo "Done."
  fi
else
  echo "Invalid color: ${PLACES_COLOR}"
  echo "Please peak one of:"
  echo "$(colors)"
  exit 1
fi
```

</p>
</details> 

### Usage

```sh
./places-color-chooser.sh --help

Places color chooser

Usage: places-color-chooser.sh [-c | --color] PLACES_COLOR [-h | --help] [-l | --list] [-n | --dry-run] [-v | --verbose]

Environment:
  PLACES_COLOR  color to change to (default: plasma)

Options:
  -c, --color=PLACES_COLOR    set the new places color (default: plasma)
  -h, --help                  show this help
  -l, --list                  list available colors
  -n, --dry-run               perform a trial run with no changes made
  -v, --verbose               increase verbosity
```

```sh
./places-color-chooser.sh --list

Available colors are:
black blue citron firebrick gold green grey highland jade lavender lime olive orange pistachio pumpkin purple red rust sapphire tomato violet white yellow
```

```sh
./places-color-chooser.sh --dry-run --color blue

bookmark-blue-missing.svg → bookmark-missing.svg
bookmarks-blue.svg → bookmarks.svg
folder-blue-activities.svg → folder-activities.svg
folder-blue-apple.svg → folder-apple.svg
folder-blue-apps.svg → folder-apps.svg
folder-blue-atom.svg → folder-atom.svg
folder-blue-cd.svg → folder-cd.svg
folder-blue-cinnamon.svg → folder-cinnamon.svg
folder-blue-desktop.svg → folder-desktop.svg
folder-blue-development.svg → folder-development.svg
folder-blue-docker.svg → folder-docker.svg
folder-blue-documents-open.svg → folder-documents-open.svg
folder-blue-documents.svg → folder-documents.svg
folder-blue-download-open.svg → folder-download-open.svg
folder-blue-downloads.svg → folder-downloads.svg
folder-blue-download.svg → folder-download.svg
folder-blue-drag-accept.svg → folder-drag-accept.svg
folder-blue-dropbox.svg → folder-dropbox.svg
folder-blue-electron.svg → folder-electron.svg
folder-blue-favorites.svg → folder-favorites.svg
folder-blue-fonts.svg → folder-fonts.svg
folder-blue-games.svg → folder-games.svg
folder-blue-github.svg → folder-github.svg
folder-blue-gitlab.svg → folder-gitlab.svg
folder-blue-git.svg → folder-git.svg
folder-blue-gnome.svg → folder-gnome.svg
folder-blue-google-drive.svg → folder-google-drive.svg
folder-blue-icons.svg → folder-icons.svg
folder-blue-image-people.svg → folder-image-people.svg
folder-blue-important.svg → folder-important.svg
folder-blue-java.svg → folder-java.svg
folder-blue-linux.svg → folder-linux.svg
folder-blue-locked.svg → folder-locked.svg
folder-blue-mac.svg → folder-mac.svg
folder-blue-mail-cloud.svg → folder-mail-cloud.svg
folder-blue-mail.svg → folder-mail.svg
folder-blue-mega.svg → folder-mega.svg
folder-blue-meocloud.svg → folder-meocloud.svg
folder-blue-mozilla.svg → folder-mozilla.svg
folder-blue-music-open.svg → folder-music-open.svg
folder-blue-music.svg → folder-music.svg
folder-blue-network.svg → folder-network.svg
folder-blue-open.svg → folder-open.svg
folder-blue-owncloud.svg → folder-owncloud.svg
folder-blue-pcloud.svg → folder-pcloud.svg
folder-blue-photo.svg → folder-photo.svg
folder-blue-pictures-open.svg → folder-pictures-open.svg
folder-blue-pictures.svg → folder-pictures.svg
folder-blue-print.svg → folder-print.svg
folder-blue-private.svg → folder-private.svg
folder-blue-publicshare-open.svg → folder-publicshare-open.svg
folder-blue-public.svg → folder-public.svg
folder-blue-recent.svg → folder-recent.svg
folder-blue-remote-open.svg → folder-remote-open.svg
folder-blue-remote.svg → folder-remote.svg
folder-blue-saved-search.svg → folder-saved-search.svg
folder-blue-script.svg → folder-script.svg
folder-blue-steam.svg → folder-steam.svg
folder-blue.svg → folder.svg
folder-blue-tar.svg → folder-tar.svg
folder-blue-templates-open.svg → folder-templates-open.svg
folder-blue-templates.svg → folder-templates.svg
folder-blue-themes.svg → folder-themes.svg
folder-blue-torrent.svg → folder-torrent.svg
folder-blue-unlocked.svg → folder-unlocked.svg
folder-blue-vbox.svg → folder-vbox.svg
folder-blue-videos-open.svg → folder-videos-open.svg
folder-blue-videos.svg → folder-videos.svg
folder-blue-video.svg → folder-video.svg
folder-blue-visiting.svg → folder-visiting.svg
folder-blue-vscode.svg → folder-vscode.svg
folder-blue-wifi.svg → folder-wifi.svg
folder-blue-wine.svg → folder-wine.svg
folder-blue-yandex-disk.svg → folder-yandex-disk.svg
user-blue-desktop.svg → user-desktop.svg
user-blue-home-open.svg → user-home-open.svg
user-blue-home.svg → user-home.svg
Nothing done.

**Reply by SylEleuth** [reply]
Dry run in example above works fine, but when I try to change colour with `--color blue` or `-c blue` I've got:
`./places-color-chooser.sh: line 52: DRYRUN: unbound variable`

**Reply by akarzim** [reply]
My bad, I forgot 2 little lines after dealing with options :face_with_peeking_eye: 

```sh
DRYRUN="${DRYRUN:-0}"
VERBOSE="${VERBOSE:-0}"
```

I updated the script in my previous comment. Sorry about that :bow:

**Reply by SylEleuth** [reply]
I've used updated script and it doesn't work. `./places-color-chooser.sh -c blue` just lists info same as during dry run but nothing changes.

**Reply by akarzim** [reply]
So sorry 🫣 The handling of boolean variables in bash confused me. It's fixed (again). And this time I double-checked and it works fine for me.

The changed lines are : 

```diff
-    if [[ ${DRYRUN} || ${VERBOSE} ]]; then
-      echo "${filename} → ${filename/-${PLACES_COLOR}/}"
-    fi
-
-    if [[ ! ${DRYRUN} ]]; then
-      ln -sfn "${filename}" "${filename/-${PLACES_COLOR}/}"
-    fi
+    if [[ ${DRYRUN} -eq 1 || ${VERBOSE} -eq 1 ]]; then
+      echo "${filename} → ${filename/-${PLACES_COLOR}/}"
+    fi
+
+    if [[ ${DRYRUN} -eq 0 ]]; then
+      ln -sfn "${filename}" "${filename/-${PLACES_COLOR}/}"
+    fi
```

Do you want a PR for this script? How about placing it in a `scripts/` directory at the root of this project?

**Reply by SylEleuth** [reply]
Script works properly now. There is one issue - `bookmark-*colour*` should create a symlink named `folder-bookmark`, not `bookmarks`.

> Do you want a PR for this script

Let's polish it a little bit first, if You don't mind.

**Reply by akarzim** [reply]
> Script works properly now. There is one issue - `bookmark-*colour*` should create a symlink named `folder-bookmark`, not `bookmarks`.

Oh! A special case. Let's deal with it.

```sh
    if [[ ${DRYRUN} -eq 1 || ${VERBOSE} -eq 1 ]]; then
      case "${filename}" in
        "bookmarks-${PLACES_COLOR}.svg")
          echo "${filename} → folder-bookmark.svg"
          ;;

        *)
          echo "${filename} → ${filename/-${PLACES_COLOR}/}"
          ;;
      esac
    fi

    if [[ ${DRYRUN} -eq 0 ]]; then
      case "${filename}" in
        "bookmarks-${PLACES_COLOR}.svg")
          ln -sfn "${filename}" "folder-bookmark.svg"
          ;;

        *)
          ln -sfn "${filename}" "${filename/-${PLACES_COLOR}/}"
          ;;
      esac
    fi
```

Script updated in previous comment :wink:

**Comment by SylEleuth** [comment]
I've made several test and it looks like it works perfectly. I have a few questions and suggestions:
1. There is a fixed number of colours and they are not going to change. Is it necessary to use "complicated" command to find all of them instead of just listing as a plain text? Maybe for the situation if someone decides to remove some colours and keep only a few.
2. `places` is the name of the directory; `folders` would be better and clearer (in display text, not variable).
3. I was thinking about usage of a dry run but couldn't find any; it's cool and nice for me or You for a tests (and I'm not sure I will use again), but for an ordinary user it's not necessary and maybe a little bit confusing (feel free to convince me to keep it).
4. Using `-v` or `--verbose` lists all changes with "Done." at the bottom.
5. I understand that `verbose` is a standard in that kind of scripts or programs. I'm just not sure if it makes sense here. It lists 128 files, which is a lot and takes significant chunk of space and even I don't go through all of it, and I can't imagine that ordinary user would too. Much better (from the user point of view) would be a progress bar (of course not in that script, it would have to be an interactive one). Or at least information about change e.g. `Changed from highland to pistachio`, or `No changes. It's already orange`.
6. Adding anything instead of the command e.g. `./places-color-chooser.sh anything` displays "Done."
7. Script has to be placed in a directory, which would make it hard for someone who installs it in root (like Arch does it). It's just a question but would it be possible to recognize where icons are located? Through scan of the folders (there are only 3 possible places) or pointing by the user?

**Reply by akarzim** [reply]
> I've made several test and it looks like it works perfectly. I have a few questions and suggestions:
> 
>     1. There is a fixed number of colours and they are not going to change. Is it necessary to use "complicated" command to find all of them instead of just listing as a plain text? Maybe for the situation if someone decides to remove some colours and keep only a few.

We could define a static set of colours, but calculating them is more robust and future-proof. Who knows, maybe a high-contrast colour or something will be added in the future; maybe someone will be happy to add their own colour to the set and use this script seamlessly!

>     2. `places` is the name of the directory; `folders` would be better and clearer (in display text, not variable).

Something like that?

```
❯ ./places-color-chooser.sh -h
Folders color chooser

Usage: places-color-chooser.sh [-c | --color] FOLDERS_COLOR [-h | --help] [-l | --list] [-n | --dry-run] [-v | --verbose]

Environment:
  FOLDERS_COLOR  color to change to (default: plasma)

Options:
  -c, --color=FOLDERS_COLOR   set the new folders color (default: plasma)
  -h, --help                  show this help
  -l, --list                  list available colors
  -n, --dry-run               perform a trial run with no changes made
  -v, --verbose               increase verbosity
```
 
>     3. I was thinking about usage of a dry run but couldn't find any; it's cool and nice for me or You for a tests (and I'm not sure I will use again), but for an ordinary user it's not necessary and maybe a little bit confusing (feel free to convince me to keep it).

Besides the advantages you mention for authors, I find it reassuring, as a user, to be able to know what will happen with a destructive script before running it for real.

>     4. Using `-v` or `--verbose` lists all changes with "Done." at the bottom.

Fair enough.

>     5. I understand that `verbose` is a standard in that kind of scripts or programs. I'm just not sure if it makes sense here. It lists 128 files, which is a lot and takes significant chunk of space and even I don't go through all of it, and I can't imagine that ordinary user would too. Much better (from the user point of view) would be a progress bar (of course not in that script, it would have to be an interactive one).

It gives me an idea! Some programs allow multiple levels of verbosity. One just have to call `-v` option multiple times. This way, we could have a non verbose mode without any output; a `-v` verbose mode which displays dots (a dot for each touched file), and a `-v -v` very verbose mode with the actual output.

It would produce this:

```sh
❯ ./places-color-chooser.sh -n
```

```sh
❯ ./places-color-chooser.sh -v      
................................................................................................................................
Done.
```

```sh
❯ ./places-color-chooser.sh -n -v -v
bookmark-plasma-missing.svg → bookmark-missing.svg
bookmarks-plasma.svg → folder-bookmark.svg
folder-plasma-activities.svg → folder-activities.svg
...
folder-plasma-yandex-disk.svg → folder-yandex-disk.svg
user-plasma-desktop.svg → user-desktop.svg
user-plasma-trash-full.svg → user-trash-full.svg
user-plasma-trash.svg → user-trash.svg
Nothing done.
```

> Or at least information about change e.g. `Changed from highland to pistachio`, or `No changes. It's already orange`.

It will take a bit more thought to work out how to follow the symbolic links to extract the destination colour. But it's an excellent idea!

>     6. Adding anything instead of the command e.g. `./places-color-chooser.sh anything` displays "Done."

Yes, all unknown arguments are silently ignored. In your example, the result will be the same as if you ran the script without arguments. I think I could find a way to raise an exception if an unknown argument is found.

>     7. Script has to be placed in a directory, which would make it hard for someone who installs it in root (like Arch does it). It's just a question but would it be possible to recognize where icons are located? Through scan of the folders (there are only 3 possible places) or pointing by the user?

Locally, I've placed this script in a `scripts/` directory at the root of the project, and the icons directory is defined in relation to this script. This way, the script can be run from anywhere, no need to be in the `scripts/` directory to run it.

But I'm working from a clone of the git repository, and have linked `Gruvbox-Plus-Dark/` to `~/.icons/Gruvbox-Plus-Dark`.

If the icon pack is installed from Arch, or Debian, chances are that the chosen pack is installed in this `~/.icons` directory, right? If that's the case, I can ask the script to look in that directory rather than the relative directory. If not, I think we can find a way to locate our target. What 3 possible locations can you think of?

**Reply by akarzim** [reply]
I've just created a pull request (https://github.com/SylEleuth/gruvbox-plus-icon-pack/pull/145) with all your requests satisfied. Each item has its own commit. It will be easier to revise and iterate on it :wink:
